stages/org.osbuild.coreos.platform

```diff
@@ -0,0 +1,108 @@
+a few steps:
+1. Locate the source of platform specific information that is provided
+   in the CoreOS filesystem tree already (the platforms.json).
+2. Copy the platforms.json file into the /boot/ partition, which is
+   sometimes used by coreos-installer.
+3. Read the platforms.json to fetch and platform specific kernel
+   arguments or grub configuration to set. These arguments/config
+   are primarily console settings.
+4. Apply any platform specific kernel arguments along with the
+   `ignition.platform.id={platform-name}` kernel argument.
+5. Apply any platform specific grub configuration.
+This stage is highly CoreOS specific and subject to change in the
+future if/when we change the way platform specific information is
+defined in our broader efforts to share more defaults with OSBuild.
+"""
+SCHEMA_2 = r"""
+"options": {
+  "additionalProperties": false,
+  "properties": {
+    "platform": {
+      "description": "The target platform name/ID",
+      "type": "string"
+    }
+  }
+},
+"devices": {
+  "type": "object",
+  "additionalProperties": true
+},
+"mounts": {
+  "type": "array"
+}
+"""
+import json
+import os
+import osbuild.api
+import shutil
+import sys
+
+from osbuild.util import append, parsing
+
+# Constants
+GRUB_CONSOLE_CFG = "/boot/grub2/console.cfg"
+PLATFORMS_JSON_DEST= "/boot/coreos/platforms.json"
+PLATFORMS_JSON_SOURCE = "/usr/share/coreos-assembler/platforms.json"
+
+def generate_console_settings_file(console_settings, file_path):
+    settings_content = "\n".join(console_settings)
+    file_content = f"""\
+# Any non-default console settings will be inserted here.
+# CONSOLE-SETTINGS-START
+{settings_content}
+# CONSOLE-SETTINGS-END
+"""
+    with open(file_path, 'w') as file:
+        file.write(file_content)
+
+
+def process_platforms_json(json_file_path, platform):
+    keys = ["grub_commands", "kernel_arguments"]
+    result = {}
+    with open(json_file_path, 'r') as file:
+        data = json.load(file)
+        if platform in data:
+            for key in keys:
+                if key in data[platform]:
+                    result[key] = data[platform][key]
+    return result
+
+
+def main(paths, tree, options):
+
+    bootpath = options.get("bootpath", "tree:///boot")
+    deployment = options.get("deployment")
+    platform = options.get("platform")
+
+    root = args["paths"]["mounts"]
+
+    platforms_source_path = root + PLATFORMS_JSON_SOURCE
+    platforms_dest_path = root + PLATFORMS_JSON_DEST
+    grub_console_cfg_path = root + GRUB_CONSOLE_CFG
+
+    kernel_arguments = [f"ignition.platform.id={platform}"]
+
+    if os.path.exists(platforms_source_path):
+
+        # Copy platforms.json to the boot partition
+        shutil.copy2(platforms_source_path, platforms_dest_path)
+        arguments = process_platforms_json(platforms_dest_path, platform)
+        if "kernel_arguments" in arguments:
+            kernel_arguments.append(arguments["kernel_arguments"])
```

Comment:
This needs to be an `extend`, not an `append`:

```suggestion
            kernel_arguments.extend(arguments["kernel_arguments"])
```

otherwise you get something like:

```
kernel arguments are: ['ignition.platform.id=qemu', ['console=tty0', 'console=ttyS0,115200n8']]
```